packages/dds/sequence/src/test/revertibles.spec.ts

```diff
@@ -0,0 +1,182 @@
+
+describe("Sequence.Revertibles with Local Edits", () => {
+	let sharedString: SharedString;
+	let dataStoreRuntime1: MockFluidDataStoreRuntime;
+	let collection: IntervalCollection<SequenceInterval>;
+	let revertibles: IntervalRevertible[];
+	const stringFactory = new SharedStringFactory();
+
+	beforeEach(() => {
+		dataStoreRuntime1 = new MockFluidDataStoreRuntime({ clientId: "1" });
+		dataStoreRuntime1.local = true;
+		sharedString = stringFactory.create(dataStoreRuntime1, "shared-string-1");
+		collection = sharedString.getIntervalCollection("test");
+		revertibles = [];
+	});
+
+	it("revert direct interval insert", () => {
+		collection.on("addInterval", (interval, local, op) => {
+			appendLocalAddToRevertibles(interval, revertibles);
+		});
+
+		sharedString.insertText(0, "hello world");
+		collection.add(0, 5, IntervalType.Simple);
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		assertIntervals(sharedString, collection, []);
+	});
+	it("revert direct interval remove", () => {
+		collection.on("deleteInterval", (interval, local, op) => {
+			appendLocalDeleteToRevertibles(sharedString, interval, revertibles);
+		});
+
+		sharedString.insertText(0, "hello world");
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id = collection.add(0, 5, IntervalType.Simple).getIntervalId()!;
+		collection.removeIntervalById(id);
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		assertIntervals(sharedString, collection, [{ start: 0, end: 5 }]);
+	});
+	it("revert direct interval change", () => {
+		collection.on("changeInterval", (interval, previousInterval, local, op) => {
+			appendLocalChangeToRevertibles(sharedString, interval, previousInterval, revertibles);
+			console.log(`${revertibles.length}`);
+		});
+
+		sharedString.insertText(0, "hello world");
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id = collection.add(0, 5, IntervalType.Simple).getIntervalId()!;
+		collection.change(id, 1, 6);
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		assertIntervals(sharedString, collection, [{ start: 0, end: 5 }]);
+	});
+	it("revert direct interval property change", () => {
+		collection.on("propertyChanged", (interval, propertyDeltas, local, op) => {
+			appendLocalPropertyChangedToRevertibles(interval, propertyDeltas, revertibles);
+		});
+
+		sharedString.insertText(0, "hello world");
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id = collection.add(0, 5, IntervalType.Simple, { foo: "one" }).getIntervalId()!;
+		collection.changeProperties(id, { foo: "two" });
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		assertIntervals(sharedString, collection, [{ start: 0, end: 5 }]);
+		const int = collection.getIntervalById(id);
+		assert.equal(int?.properties.foo, "one");
+	});
+});
+describe("Sequence.Revertibles with Remote Edits", () => {
```

Comment:
Remote edit cases should also include cases where the remote edits were interval edits. e.g. local change properties, remote change properties. I can help brainstorm if you want.